ggml-cuda.cu

```diff
@@ -1174,16 +1175,12 @@ static __device__ void convert_f16(const void * vx, const int ib, const int iqs,
 static __global__ void quantize_q8_1(const float * __restrict__ x, void * __restrict__ vy, const int k) {
     const int i = blockDim.x*blockIdx.x + threadIdx.x;
 
-    if (i >= k) {
-        return;
-    }
```

Comment:
I imagine this is still needed if `MATRIX_ROW_PADDING != CUDA_QUANTIZE_BLOCK_SIZE`. Maybe this could be done conditionally to improve performance:
```C++
#if MATRIX_ROW_PADDING != CUDA_QUANTIZE_BLOCK_SIZE
    if (i >= k) {
        return;
    }
#endif
```

Reply:
The reason I removed the check is that the padded area needs to be filled with zeros so that the `mul_mat_vec_q` kernel can process those areas unconditionally without affecting the result of the scalar product; effectively it will access elements from the next row (or the padding at the end) but multiply them with 0 so those elements have no effect on results. If the check is re-added then you would have to add a call to `cudaMemset` to zero the padded region of the vector and that would very likely have worse performance.

How about this: instead of passing just one value for `k`, pass two values. The first value specifies the number of elements in the unquantized vector and can be used to determine whether to use `0.0f` instead of vector data. The second value specifies the number of elements in the quantized and padded vector and determines when threads should immediately return.